### PR TITLE
ci fix 8

### PR DESCRIPTION
**Pull request by cyx-6**

### CODE REVIEW

File: python/tvm/script/parser/tir/parser.py [hunk 1]
```diff
@@ -199,16 +199,20 @@ def visit_with(self: Parser, node: doc.With) -> None:
...
+                if node.returns is not None:
+                    ret_type = self.eval_expr(node.returns)
+                    if callable(ret_type):
+                        ret_type = PrimType(ret_type().dtype)
+                    T.func_ret(ret_type)
+                with self.with_dispatch_token("tir"):
+                    self.visit(node.args)
+                    self.visit_body(node.body)
+        except Exception as e:
+            self.report_error(node, str(e))
```
Comment (cyx-6) [inline python/tvm/script/parser/tir/parser.py]: line 214 may cause SegFault

Comment (cyx-6) [reply python/tvm/script/parser/tir/parser.py]:
```python
@T.prim_func
def opaque_access_during_complete(a: T.handle) -> None:  # error
    A = T.match_buffer(a, (16, 16), "float32")
    for i, j in T.grid(16, 16):
        with T.block():
            T.evaluate(T.call_extern("dummy_extern_function", A.data, dtype="int32"))
```

File: include/tvm/tir/ir_builder.h [hunk 2]
```diff
@@ -29,6 +29,7 @@ namespace tir {
 
 using tvm::runtime::NDArray;
 using tvm::tir::Buffer;
+using tvm::tir::IterVar;
```
Comment (junrushao) [inline include/tvm/tir/ir_builder.h]: remove this - not used

File: python/tvm/script/parser/tir/parser.py [hunk 3]
```diff
@@ -199,16 +199,20 @@ def visit_with(self: Parser, node: doc.With) -> None:
...
-            T.func_name(node.name)
-            if node.returns is not None:
-                ret_type = self.eval_expr(node.returns)
-                if callable(ret_type):
-                    ret_type = PrimType(ret_type().dtype)
-                T.func_ret(ret_type)
-            with self.with_dispatch_token("tir"):
-                self.visit(node.args)
-                self.visit_body(node.body)
+        try:
```
Comment (junrushao) [inline python/tvm/script/parser/tir/parser.py]: error handling logic should be moved to around this line: https://github.com/apache/tvm/blob/ir-builder/python/tvm/script/parser/parser.py#L146.

Similarly, here: https://github.com/apache/tvm/blob/ir-builder/python/tvm/script/parser/parser.py#L152; and here: https://github.com/apache/tvm/blob/ir-builder/python/tvm/script/parser/parser.py#L126

File: include/tvm/tir/ir_builder.h (hunk 2 above)
Comment (cyx-6) [reply include/tvm/tir/ir_builder.h]: move it to `ir_builder.cc`?
it is used there

Comment (junrushao) [reply include/tvm/tir/ir_builder.h]: got it

Comment (junrushao) [reply include/tvm/tir/ir_builder.h]: go ahead

**Review by junrushao — APPROVED** [review]

**Pull request merged**